### PR TITLE
feat(core,schemas): implement social verification experience API endpoints

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Implement the social verification class and social verification endpoints.

- `experience/verification/social/:connectorId/authorization-uri`
- `experience/verification/social/:connectorId/verify`

These two endpoints handle the social verification creation and verification logic. 

This PR also includes mockSocialConnector updates, and adds the `connectorSession` write and read logic. So in our integration tests, the `connectorSession`  write and retrieve logic can be tested. 

## Testing

integration tests added 

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/experience/classes/interaction-session.ts [hunk 1]
```diff
@@ -91,7 +92,8 @@ export default class InteractionSession {
           status: 404,
         },
         {
-          identifier: verificationRecord.identifier.value,
+          identifier:
+            'identifier' in verificationRecord ? verificationRecord.identifier : undefined,
```
Comment (simeng-li) [inline packages/core/src/routes/experience/classes/interaction-session.ts:96]: This piece of logic will be refactored in the coming next PR, please ignore it. See the comments above.

File: packages/core/src/routes/experience/classes/verifications/social-verification.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    const {
+      metadata: { target },
+    } = await socials.getConnector(this.connectorId);
+
+    const user = await findUserByIdentity(target, this.socialUserInfo.id);
+
+    return user ?? undefined;
+  }
+
+  async findRelatedUserBySocialIdentity(): ReturnType<typeof socials.findSocialRelatedUser> {
```
Comment (wangsijie) [inline packages/core/src/routes/experience/classes/verifications/social-verification.ts:163]: nit: add some comments to this method about the difference between this and `findUserBySocialIdentity`

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/routes/experience/classes/verifications/social-verification.ts (hunk 2 above)
Comment (simeng-li) [reply packages/core/src/routes/experience/classes/verifications/social-verification.ts:166]: Added

**Review by wangsijie — APPROVED** [review]

**Review by darcyYe — APPROVED** [review]
Changes on mock social connectors looks good to me

**Pull request merged**